### PR TITLE
add_callback

### LINKED ISSUE #228: Allow to specify stream=True for callback responses
According to https://github.com/getsentry/responses/blob/master/responses.py#L355 default stream value for callback responses is False.
Also there is no way to override this value using public method add_callback https://github.com/getsentry/responses/blob/master/responses.py#L510
As a result this doesn't work:
```
        def callback(request):
            return 200, {}, 'content'
        responses.add_callback(
            'GET',
            'https://domain.com/',
            callback
        )
        response = requests.get('https://domain.com/', stream=True)
        content = response.raw.read()
....
    response.raw.read()
  File "/usr/lib/python3.6/site-packages/urllib3/response.py", line 380, in read
    data = self._fp.read()
ValueError: I/O operation on closed file.
```
Looks like it can be fixed by adding stream argument to add_callback method

As an workaround one can add a match manually using context manager:
```
        def callback(request):
            return 200, {}, 'content'
        with responses.RequestsMock() as requests_mock:

            requests_mock._matches.append(responses.CallbackResponse(
                'GET',
                'https://domain.com/',
                callback, stream=True
            ))
            response = requests.get('https://domain.com/', stream=True)
            content = response.raw.read()
```

### PR DESCRIPTION
**Pull request by beliaev-maksim**
added match to add_callback
replace match default to tuple since it is not a good idea to have mutable objects in the signature

after this PR we can close PR #268 since `stream` now should be set via request and all other interfaces of `CallbackResponse` are added via current PR

Closes #228

### CODE REVIEW

File: responses/__init__.pyi [hunk 1]
```diff
@@ -199,7 +200,7 @@ class _Add(Protocol):
         auto_calculate_content_length: bool = ...,
         adding_headers: Optional[Mapping[str, str]] = ...,
         match_querystring: bool = ...,
-        match: List[Any] = ...,
+        match: Iterable[Callable[[Any], Callable[..., Any]]] = ...,
```
[the remark this answers is not in the document]
Comment (beliaev-maksim) [reply responses/__init__.pyi]: @markstory 
added `MatcherIterable`

File: responses/__init__.py [hunk 2]
```diff
@@ -659,7 +659,13 @@ def upsert(self, method_or_response=None, url=None, body="", *args, **kwargs):
...
 
     def add_callback(
-        self, method, url, callback, match_querystring=False, content_type="text/plain"
+        self,
+        method,
+        url,
+        callback,
+        match_querystring=False,
+        content_type="text/plain",
+        match=(),
```
[the remark this answers is not in the document]
Comment (beliaev-maksim) [reply responses/__init__.py:678]: added two :D

**Comment by markstory** [comment]
Thank you 👍

**Comment by beliaev-maksim** [comment]
@markstory 
thanks for review!

**Pull request merged**